concent_api/common/decorators.py

```diff
@@ -169,9 +171,45 @@ def wrapper(request, client_message, client_public_key, *args, **kwargs):
             try:
                 if database_name is not None:
                     sid = transaction.savepoint(using=database_name)
-                response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
+                try:
+                    response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
+                except DatabaseError as e:
+                    # This section is to catch Database errors caused by multiple requests about one Subtask.
+                    # This requests are waiting and retrying.
+                    log_string_message(
+                        logger,
+                        f'DatabaseError during process client request. Client public key: {client_public_key}.'
+                        f'Error: {e}'
+                    )
+                    sleep_time = settings.INITIAL_VIEW_RETRY_DELAY
+                    subtask_got_successfully = False
+                    response_from_view = None
+
+                    for counter in range(settings.MAX_VIEW_RETRIES):
+                        try:
+                            time.sleep(sleep_time)
+                            sleep_time *= 2
+                            response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
+                            subtask_got_successfully = True
+                            break
+                        except DatabaseError as e:
+                            log_string_message(
+                                logger,
+                                f'DatabaseError during process client request. Client public key: {client_public_key}.'
+                                f'Number of retries {counter+1}/{settings.MAX_VIEW_RETRIES}'
+                                f'Error: {e}'
+                            )
+                    if not subtask_got_successfully:
```

Comment:
this is not necessary - please use `for ... else` instead of `subtask_got_successfully` flag

Reply:
http://book.pythontips.com/en/latest/for_-_else.html

Reply:
Right... java doesn't have such construction. Will remember, but in this case, after move to separate function not necessary any more